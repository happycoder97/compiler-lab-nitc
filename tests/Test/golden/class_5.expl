class A         
{
  int m1() {
    write("A M1");
  }

  int m2() {
    write("A M2");
  }

  int m3() {
    write("A M3");
  }
}

class B extends A
{
  int m1() {
    write("B M1");
  }

  int m3() {
    write("B M3");
  }
}

A a;
B b;
A a1;

int main() {
    initialize();
    a = new(A);
    b = new(B);
    a1 = new(B);
    a.m1();
    a.m2();
    a.m3();
    b.m1();
    b.m2();
    b.m3();
    a1.m1();
    a1.m2();
    a1.m3();
}
