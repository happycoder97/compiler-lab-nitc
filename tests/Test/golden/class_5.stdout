A M1
A M2
A M3
B M1
A M2
B M3
B M1
A M2
B M3
